src/components/SubMenu.vue

```diff
@@ -28,23 +37,36 @@ export default defineComponent({
       type: Object as PropType<MenuItem>,
       required: true,
     },
+    anchor: Object as PropType<QMenu['anchor']>,
+    self: Object as PropType<QMenu['self']>,
     level: { type: Number, required: true },
   },
-  setup(props) {
+  emits: ['close'],
+  setup(props, { emit }) {
+    const qMenu = ref<QMenu>();
     const locale = ref<string>('en');
 
     function getLocalizedTitle(menu: MenuItem): string {
       const titleKey = `title_${locale.value}`;
       return menu.attributes[titleKey as keyof MenuItemAttributes] as string;
     }
 
+    function close(menu?: MenuItem) {
```

Comment:
Here we are checking if the menu has submenus; if it does, abort the close operation.

Reply:
Amazing. Big Thanks.